docs/scripting-element-object.rst

```diff
@@ -0,0 +1,550 @@
+    end
+
+
+.. _splash-element-png:
+
+element:png
+-----------
+
+Return a screenshot of the element in PNG format
+
+**Signature:** ``ok, shot = element:png{width=nil, height=nil, scale_method='raster', pad=0}``
+
+**Parameters:**
+
+* width - optional, width of a screenshot in pixels;
+* height - optional, height of a screenshot in pixels;
+* scale_method - optional, method to use when resizing the image, ``'raster'``
+  or ``'vector'``;
+* pad - optional, integer or ``{left, top, right, bottom}`` values of padding
+
+**Returns:** ``ok, shot`` pair. If ``ok`` is nil then error happened during the
+function call; ``shot`` provides an information about error type; otherwise
+``shot`` is a PNG screenshot data, as a :ref:`binary object <binary-objects>`.
+When the result is empty (e.g. if the element is not visible) ``nil`` is returned.
+
+**Async:** no.
+
+*pad* parameter sets the padding of the resulting image. If it is a single integer then the
+padding from all sides will be equal. If the value of the padding is positive the resulting screenshot
+will be expanded by the specified amount of pixes. And if the value of padding is negative the resulting
+screenshot will be shrunk by the specified amount of pixes.
+
+See more in :ref:`splash-png`.
+
+
+.. _splash-element-jpeg:
+
+element:jpeg
+------------
+
+Return a screenshot of the element in JPEG format
+
+**Signature:** ``ok, shot = element:jpeg{width=nil, height=nil, scale_method='raster', quality=75, region=nil, pad=0}``
+
+**Parameters:**
+
+* width - optional, width of a screenshot in pixels;
+* height - optional, height of a screenshot in pixels;
+* scale_method - optional, method to use when resizing the image, ``'raster'``
+  or ``'vector'``;
+* quality - optional, quality of JPEG image, integer in range from ``0`` to ``100``;
+* pad - optional, integer or ``{left, top, right, bottom}`` values of padding
+
+**Returns:** ``ok, shot`` pair. If ``ok`` is nil then error happened during the
+function call; ``shot`` provides an information about error type; otherwise
+``shot`` is a JPEG screenshot data, as a :ref:`binary object <binary-objects>`.
+When the result is empty (e.g. if the element is not visible) ``nil`` is returned.
+
+**Async:** no.
+
+*pad* parameter sets the padding of the resulting image. If it is a single integer then the
+padding from all sides will be equal. If the value of the padding is positive the resulting screenshot
+will be expanded by the specified amount of pixes. And if the value of padding is negative the resulting
+screenshot will be shrunk by the specified amount of pixes.
+
+See more in :ref:`splash-jpeg`.
+
+
+.. _splash-element-visible:
+
+element:visible
+---------------
+
+Check whether the element is visible.
+
+**Signature:** ``visible = element:visible()``
+
+**Returns:** ``visible`` indicated whether
+the element is visible.
+
+**Async:** no.
+
+
+.. _splash-element-fetch-text:
+
+element:fetch_text
```

Comment:
I don't quite like the `fetch_text` name because `fetch` usually implies some async action, e.g. a HTTP request. What about `element:get_text()` or just `element:text()`?

By the way, it looks like some of the methods use `get_` prefix, and some of them don't. Is it intentional, it there a pattern?